lock.go

```diff
@@ -18,13 +18,29 @@ import (
 const LockName = "Gopkg.lock"
 
 type Lock struct {
-	Memo []byte
-	P    []gps.LockedProject
+	SolveMeta SolveMeta
+	P         []gps.LockedProject
+}
+
+type SolveMeta struct {
+	Memo            []byte
+	AnalyzerName    string
+	AnalyzerVersion int
+	SolverName      string
+	SolverVersion   int
 }
 
 type rawLock struct {
-	Memo     string             `toml:"memo"`
-	Projects []rawLockedProject `toml:"projects"`
+	SolveMeta solveMeta          `toml:"solve-meta"`
+	Projects  []rawLockedProject `toml:"projects"`
+}
+
+type solveMeta struct {
+	Memo            string `toml:"inputs-hash"`
```

Comment:
Yeah I kinda flip-flopped on this. There are a bunch of different similar names that get used for this - both the method that does it, and the name of fields that store it. (This mirrors the common conflation of "hashing" with "hash digest"). I didn't change it only because I kinda didn't want to start unpacking that can of worms...though still, yes, it would be incrementally better to change it here now and then change it again later than have the incongruity.